jose/parse_test.go

```diff
@@ -33,6 +36,24 @@ const (
 	octKey
 )
 
+type wrapSigner struct {
+	crypto.Signer
+}
+
+func (w wrapSigner) Public() crypto.PublicKey {
+	if w.Signer == nil {
+		return nil
+	}
+	return w.Signer.Public()
+}
+
+func (w wrapSigner) Sign(r io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
+	if w.Signer == nil {
+		return nil, errors.New("not implemented")
+	}
```

Comment:
Since `wrapSigner` is only used in tests, you could make it take `t *testing.T`, and fail immediately when `Signer` is nil. Then you can remove some test cases that are only testing this test struct's functioning.

Reply:
There are specific cases where I don't want the test to fail. For example, when I create an opaque signer with `NewOpaqueSigner(wrapSigner{})`, this method will end up calling `signer.Public()`, I don't care about the output, but I don't want the method to fail, I want only that the opaqueSigner is not able to recognize the type of the key.